server/src/Main.hs

```diff
@@ -173,28 +320,43 @@ beginLogin db pending = do
           WA.coaExtensions = Nothing
         }
 
+  -- Send credential options to the client
   Scotty.liftAndCatchIO $ TIO.putStrLn $ "Login begin => " <> jsonText options
   Scotty.json $ WA.encodeCredentialOptionsAuthentication options
+  where
+    mkCredentialDescriptor :: WA.CredentialEntry -> WA.CredentialDescriptor
+    mkCredentialDescriptor WA.CredentialEntry {WA.ceCredentialId, WA.ceTransports} =
+      WA.CredentialDescriptor
+        { WA.cdTyp = WA.CredentialTypePublicKey,
+          WA.cdId = ceCredentialId,
+          WA.cdTransports = Just ceTransports
+        }
 
-completeLogin :: WA.Origin -> WA.RpIdHash -> Database.Connection -> PendingOps -> Scotty.ActionM ()
+-- | Receives the credential from the client and perform the authentication
+-- ceremony. If the credential is valid the user is logged in.
```

Comment:
Same here, the authentication ceremony is being completed, but this function only executes the relying party operation (which is part of the ceremony).

Technically the ceremony still isn't done after this (see the definition of a [ceremony](https://www.w3.org/TR/webauthn-2/#ceremony)), but it's completed for at least the server, after this only the client/user is involved.